app/models/course/condition.rb

```diff
@@ -5,4 +5,6 @@ class Course::Condition < ActiveRecord::Base
   belongs_to :conditional, polymorphic: true
 
   delegate :satisfied_by?, to: :actable
+
+  NAMES = ['achievement', 'assessment', 'level']
```

Comment:
Hmm, but the specific conditions are not technically subclasses of `Course::Condition`. They are declared as `class Course::Condition::Achievement < ActiveRecord::Base`. This is how acts_as models are declared. I'm not sure if declaring as subclass gives any side effect.